tests/providers/amazon/aws/hooks/test_ecr.py

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from unittest import mock
+
+import boto3
+import pytest
+from moto import mock_ecr
+from moto.core import DEFAULT_ACCOUNT_ID
+
+from airflow.providers.amazon.aws.hooks.ecr import EcrHook
+
+
+@pytest.fixture
+def patch_hook(monkeypatch):
+    """Patch hook object by dummy boto3 ECR client."""
+    ecr_client = boto3.client("ecr")
+    monkeypatch.setattr(EcrHook, "conn", ecr_client)
+    yield
+
+
+@mock_ecr
+class TestEcrHook:
+    def test_service_type(self):
+        """Test expected boto3 client type."""
+        assert EcrHook().client_type == "ecr"
+
+    @pytest.mark.parametrize(
+        "accounts_ids",
+        [
+            pytest.param("", id="empty-string"),
+            pytest.param(None, id="none"),
+            pytest.param([], id="empty-list"),
+        ],
+    )
```

Comment:
This might be my own ignorance, but why bother with the pytest.param and setting an id?  I've always done this as 

`@pytest.mark.parametrize("accounts_ids", ["", None, []])`

Reply:
It is more about how it looks in `pytest` output and in PyCharm/IDEA.
More refers how `pytest` work with empty string and objects.

---

```python
@pytest.mark.parametrize("accounts_ids", ["", None, []])
def test_sample(accounts_ids):
    pass
```
```console
============================= test session starts ==============================
collecting ... collected 3 items

test_docker.py::test_sample[] 
test_docker.py::test_sample[None] 
test_docker.py::test_sample[accounts_ids2] 

============================== 3 passed in 0.12s ===============================
```
![image](https://user-images.githubusercontent.com/3998685/207433355-09909185-0d4e-4373-af20-8983181a93f1.png)

---

```python
@pytest.mark.parametrize(
    "accounts_ids",
    [
        pytest.param("", id="empty-string"),
        pytest.param(None, id="none"),
        pytest.param([], id="empty-list"),
    ],
)
def test_sample(accounts_ids):
    pass
```
```console
============================= test session starts ==============================
collecting ... collected 3 items

test_docker.py::test_sample[] 
test_docker.py::test_sample[None] 
test_docker.py::test_sample[accounts_ids2] 

============================== 3 passed in 0.12s ===============================
```
![image](https://user-images.githubusercontent.com/3998685/207433529-ca64fe1c-ba52-463a-89de-9e977fab6204.png)

Reply:
And predictable cli command for run test for single parameter. It's useful during debagging when you change something and just one parameter stop working

```console
❯ pytest path/to.py::TestClassName:test_sample[empty-list] 
```

Reply:
Interesting.  Pretty sure the second block of terminal output is a mistaken repost of the first output, but I get what you are saying.  I don't think I've ever seen that used.  :+1:

Reply:
Yeah... Update it